packages/cactus-plugin-keychain-memory/README.md

```diff
@@ -1,9 +1,41 @@
 # `@hyperledger/cactus-plugin-keychain-memory`
 
-> TODO: description
+## Prometheus Exporter
 
-## Usage
+This class creates a prometheus exporter, which scraps the transactions (total transaction count) for the use cases incorporating the use of Fabric connector plugin.
 
+
+### Usage
+The prometheus exporter object is initialized in the `PluginKeychainMemory` class constructor itself, so instantiating the object of the `PluginKeychainMemory` class, gives access to the exporter object.
+You can also initialize the prometheus exporter object seperately and then pass it to the `IPluginKeychainMemoryOptions` interface for `PluginKeychainMemory` constructor.
+
+`getPrometheusExporterMetricsEndpointV1` function returns the prometheus exporter metrics, currently displaying the total key count, which currently increments everytime the `set()` and `delete()` method of the `PluginKeychainMemory` class is called.
```

Comment:
> the total key count, which currently increments everytime the `set()` and `delete()` method of the `PluginKeychainMemory` class is called

@jagpreetsinghsasan Hmm, does that mean that if I do

```typescript
keychain.set("x", "1");
keychain.set("x", "2");
keychain.set("x", "3");
```

Then the total key count metric will show 3 instead of 1? If yes then I'd recommend changing the code so that it doesn't increment unconditionally the total key count metric but instead that it always sets the metric value to the key count of the underlying backend storage (which is just the simple `Map` instance for the in-memory plugin IIRC).
The above would ensure that every time the set method is called the total key count is updated to the correct value even if someone is overwriting the same key via multiple set operations.

See this on how to get the key count easily from the map instance: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map/size

Reply:
True. I forgot this scenario.
I will get the key count instead for the same.